### ISSUE TITLE
CondaEnvironment should use conda activate, not a manual PATH prepending

### ISSUE THREAD

**Issue by asmeurer** [Microsoft/azure-pipelines-tasks#8504]
The CondaEnvironment uses a manual activate routine, that prepends the PATH and sets the conda environment variables https://github.com/Microsoft/vsts-tasks/blob/6fae1bdf535512ecf21495fabcdc777cb7fc1c0c/Tasks/CondaEnvironmentV1/conda_internal.ts#L129

However, this is insufficient, as conda also sources activation scripts that may be included with some packages. For my use-case this breaks the `sage` package from `conda forge`, which sets some sage environment variables (see https://github.com/conda-forge/sagelib-feedstock/blob/master/recipe/activate/activate.sh). 

Instead of trying to do the activation manually, it should reuse conda's own logic for this and source conda's activate.sh.

**Comment by brcrista** [comment]
You're right, that's what the task should do.  The issue is that each build step runs in its own process.  So if we just ran `activate`, any environment state wouldn't persist in later steps.

We do have a way to pass environment between build steps, but it's something the task has to do.  What we need to do is basically implement "sourcing" logic for the task, where we run `activate` and any environment context written by it or a delegated script gets picked up and propagated.

So this is a known limitation of the task and is on our backlog.  It looks like the workaround you found is the right one though.

**Comment by brcrista** [comment]
For anyone else out there who has this issue, scripting is also an option.  You just have a script where you run `activate` and then everything else you need.  Or, you can still break a job up into individual steps (say you want that finer-grained passed / failed status), but they need to be in script and you need to re-run `activate` at the top of each one.

**Comment by asmeurer** [comment]
FWIW I've found it annoying even in other contexts that environment variables aren't shared across tasks.

**Comment by brcrista** [comment]
I hadn't noticed before, but it looks like you're right -- on Windows, activating the environment at the beginning of a script causes the rest of the script not to be executed.  I tested this on Hosted macOS and Hosted Ubuntu also and they don't have this problem.

Thanks for sharing your workaround.  Let me see what I can come up with here.

**Comment by brcrista** [comment]
@mtaron my mistake, you have to `call activate <env>` -- see https://github.com/conda/conda/issues/794

I'll admit, Batch scripting is one of my lesser strengths.  I'll fix the doc page.

**Comment by jenshnielsen** [comment]
In the newest version of `conda` i.e. with `updateConda: true` and conda 4.6.3 this seems to also break loading of the sqlite3 dynamic library and in turn the whole of the python sqlite3 module.

**Comment by brcrista** [comment]
We've decided to deprecate the Conda Environment task for the next release (see #9573).  Instead, we recommend invoking the `conda` CLI directly.  The task made sense when we were still focused on the old "Web designer"-based build definitions that lean heavily on tasks.  Today with YAML, script is much easier.

I created a guide with some YAML snippets here: https://aka.ms/pipelines-anaconda-guide.  Feedback welcome!
